diff -Naur pyxmpp-0.3.orig/pyxmpp/jid.py pyxmpp-0.3/pyxmpp/jid.py
--- pyxmpp-0.3.orig/pyxmpp/jid.py	2003-11-09 15:41:16.000000000 +0100
+++ pyxmpp-0.3/pyxmpp/jid.py	2004-01-19 11:30:20.981964512 +0100
@@ -25,7 +25,7 @@
 
 node_invalid_re=re.compile(ur"[" u'"' ur"&'/:<>@\s\x00-\x19]",re.UNICODE)
 resource_invalid_re=re.compile(ur"[\s\x00-\x19]",re.UNICODE)
-domain_invalid_re=re.compile(r"[^-a-zA-Z]")
+domain_invalid_re=re.compile(r"[^-a-zA-Z0-9]")
 
 
 class JIDError(ValueError):
